omegaconf/grammar_visitor.py

```diff
@@ -246,20 +247,41 @@ def visitSequence(
     ) -> Generator[Any, None, None]:
         from ._utils import _get_value
 
-        assert ctx.getChildCount() >= 1  # element (COMMA element)*
+        # (element (COMMA element?)*) | (COMMA element?)+
+        assert ctx.getChildCount() >= 1
+
+        def empty_str_warning() -> None:
+            txt = ctx.getText()
+            warnings.warn(
```

Comment:
Add a comment:
```
# DEPRECATED: remove in 2.2
```

This is not a practice I follow yet in OmegaConf but I think we should.
it makes it easier to spot those later.

Reply:
Done in 9ecac27

Also adds the missing tests and news I mentioned in the chat :p